### PR TITLE
Nukes recursive effects

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Nukes `/datum/effect_system/trail_follow` since it relies on so much recursion that its actually painful. Recursive loops outside of the main MC "thread" are bad, and these shouldnt be on a loop anyway, especially when we have signals.

WIP since I need to remove it from the other places too, pods were just the starting line. 

## Why It's Good For The Game
See above

## Changelog
:cl: AffectedArc07
tweak: Space trail effects (EG: Spacepods) are now a lot more sane code-side 
/:cl:

### CODE REVIEW

File: code/game/objects/effects/effect_system.dm [hunk 1]
```diff
@@ -1,1289 +0,0 @@
-/* This is an attempt to make some easily reusable "particle" type effect, to stop the code
```
Comment (AffectedArc07) [inline code/game/objects/effects/effect_system.dm:1]: This is a file which has been unticked for eons and is mostly dupe code

File: code/modules/spacepods/spacepod.dm [hunk 2]
```diff
@@ -124,6 +119,33 @@
...
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, /obj/spacepod/.proc/create_trail)
+
+/obj/spacepod/proc/create_trail()
+	var/turf/T = get_turf(src)
+	var/atom/oldposition
+	var/atom/oldloc
+	switch(dir)
+		if(NORTH)
+			oldposition = T
+			oldposition = get_step(oldposition, SOUTH)
```
Comment (AffectedArc07) [inline code/modules/spacepods/spacepod.dm:131]: Yes I will also clean these up once I finish this PR

File: code/game/objects/effects/effect_system/effects_other.dm [hunk 3]
```diff
@@ -1,140 +1,15 @@
...
-					start()
-			currloc = T
+	anchored = TRUE
+
+/obj/effect/particle_effect/ion_trails/Initialize(mapload, targetdir)
+	. = ..()
+	dir = targetdir
+	flick("ion_fade", src)
+	icon_state = null
+	QDEL_IN(src, 20)
```
Comment (dearmochi) [inline code/game/objects/effects/effect_system/effects_other.dm:12]:
```suggestion
	QDEL_IN(src, 2 SECONDS)
```

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 4]
```diff
@@ -176,6 +175,13 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 	diag_hud_set_borgcell()
 	scanner = new(src)
 	scanner.Grant(src)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, /mob/living/silicon/robot/.proc/create_trail)
+
+/mob/living/silicon/robot/proc/create_trail(datum/source, atom/OldLoc, Dir, forced)
+	if(ionpulse_on)
+		var/turf/T = get_turf(OldLoc)
+		if(!has_gravity(T))
+			new /obj/effect/particle_effect/ion_trails(T, Dir)
```
Comment (dearmochi) [inline code/modules/mob/living/silicon/robot/robot.dm:184]:
```suggestion
/mob/living/silicon/robot/proc/create_trail(datum/source, atom/oldloc, dir, forced)
	if(ionpulse_on)
		var/turf/T = get_turf(oldloc)
		if(!has_gravity(T))
			new /obj/effect/particle_effect/ion_trails(T, dir)
```

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 5]
```diff
@@ -176,6 +175,13 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 	diag_hud_set_borgcell()
 	scanner = new(src)
 	scanner.Grant(src)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, /mob/living/silicon/robot/.proc/create_trail)
```
Comment (dearmochi) [inline code/modules/mob/living/silicon/robot/robot.dm:178]:
```suggestion
	RegisterSignal(src, COMSIG_MOVABLE_MOVED, .proc/create_trail)
```

File: code/modules/mob/living/simple_animal/hostile/retaliate/drone.dm [hunk 6]
```diff
@@ -29,16 +29,18 @@
...
 	del_on_death = 1
-	var/datum/effect_system/trail_follow/ion/ion_trail
 	var/passive_mode = TRUE // if true, don't target anything.
 
 /mob/living/simple_animal/hostile/malf_drone/Initialize(mapload)
 	. = ..()
-	ion_trail = new
-	ion_trail.set_up(src)
-	ion_trail.start()
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, /mob/living/simple_animal/hostile/malf_drone/.proc/create_trail)
```
Comment (dearmochi) [inline code/modules/mob/living/simple_animal/hostile/retaliate/drone.dm:36]:
```suggestion
	RegisterSignal(src, COMSIG_MOVABLE_MOVED, .proc/create_trail)
```

File: code/modules/mob/living/simple_animal/hostile/retaliate/drone.dm [hunk 7]
```diff
@@ -29,16 +29,18 @@
...
 	update_icons()
 
+/mob/living/simple_animal/hostile/malf_drone/proc/create_trail(datum/source, atom/OldLoc, Dir, forced)
+	var/turf/T = get_turf(OldLoc)
+	if(!has_gravity(T))
+		new /obj/effect/particle_effect/ion_trails(T, Dir)
```
Comment (dearmochi) [inline code/modules/mob/living/simple_animal/hostile/retaliate/drone.dm:42]:
```suggestion
/mob/living/simple_animal/hostile/malf_drone/proc/create_trail(datum/source, atom/oldloc, dir, forced)
	var/turf/T = get_turf(oldloc)
	if(!has_gravity(T))
		new /obj/effect/particle_effect/ion_trails(T, dir)
```

File: code/modules/spacepods/spacepod.dm [hunk 8]
```diff
@@ -124,6 +119,29 @@
 	cargo_hold.max_w_class = 5		//fit almost anything
 	cargo_hold.max_combined_w_class = 0 //you can optimize your stash with larger items
 	START_PROCESSING(SSobj, src)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, /obj/spacepod/.proc/create_trail)
```
Comment (dearmochi) [inline code/modules/spacepods/spacepod.dm:122]:
```suggestion
	RegisterSignal(src, COMSIG_MOVABLE_MOVED, .proc/create_trail)
```

**Review by dearmochi — COMMENTED** [review]
Few things

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 4 above)
Comment (AffectedArc07) [reply code/modules/mob/living/silicon/robot/robot.dm:184]: I will do this for `oldloc`, but `dir` is keeping a different name to aid diffrentiation

**Review by dearmochi — APPROVED** [review]

**Pull request merged**